# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GenerateStatsPhases.py
# Commit: 9882c0d924b8
# Full Hash: 9882c0d924b8ad13e1e0828f9e3860244ad6f33f
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2023-07-08 09:31:48
# Regressor Bug: 1837620
# File Overlap Count: 1
# Description:
#   Bug 1837620 - Part 6: Make edges for multiple shape guard weak too r=sfink
#   
#   To make the edges for the multiple shape guard weak we need a way to find and
#   update the objects used to hold these edges. The simplest way to do this is to
#   add a vector in the zone.  I don't believe that there are not huge numbers of
# ==============================================================================

diff -r ba12c0cbb7a4 -r 9882c0d924b8 js/src/gc/GenerateStatsPhases.py
--- a/js/src/gc/GenerateStatsPhases.py	Fri Jul 07 17:05:44 2023 +0000
+++ b/js/src/gc/GenerateStatsPhases.py	Fri Jul 07 17:05:44 2023 +0000
@@ -186,6 +186,7 @@
                     addPhaseKind("SWEEP_COMPRESSION", "Sweep Compression Tasks", 62),
                     addPhaseKind("SWEEP_WEAKMAPS", "Sweep WeakMaps", 63),
                     addPhaseKind("SWEEP_UNIQUEIDS", "Sweep Unique IDs", 64),
+                    addPhaseKind("SWEEP_WEAK_POINTERS", "Sweep Weak Pointers", 81),
                     addPhaseKind(
                         "SWEEP_FINALIZATION_OBSERVERS",
                         "Sweep FinalizationRegistries and WeakRefs",